branchLayout/api/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -29,6 +30,28 @@ private static List<IBranchLayoutEntry> collectEntriesRecursively(IBranchLayoutE
     return entryByName.get(branchName).getOrNull();
   }
 
+  @Override
+  public @Nullable IBranchLayoutEntry findNextEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    if (currentIndex > -1 && currentIndex + 1 < entriesOrderedList.length()) {
+      @LTLengthOf("entriesOrderedList") int nextIndex = currentIndex + 1;
+      return findEntryByName(entriesOrderedList.get(nextIndex));
+    }
+    return null;
+  }
+
+  @Override
+  public @Nullable IBranchLayoutEntry findPreviousEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    if (currentIndex > 0 && currentIndex < /* added for preventing a type error */entriesOrderedList.length() - 1) {
```

Comment:
hmmm 🤔 what do you mean by `/* added for preventing a type error */` (there is missing space after the comment)? 

why the upper bound here is decreased by 1? It should be just `currentIndex < entriesOrderedList.length()`, am I right?

Reply:
it is for making our checker understand that the value passed to `.get()` is within the allowed bounds :(

Reply:
ok, but now for branch layout

```
master
develop
```

```
findPreviousBranch("develop") == null
```

Reply:
Hmmm okay let's analyze that case... won't `currentIndex` for `develop` be `1`? with `entriesOrderedList.length()` being 2, the `if`'s condition isn't true 🤔 so I'd indeed see `null` being returned here... something's fishy here 🐟

Reply:
Locally, the build was successful with `-PrunAllCheckers`